test/runtests.jl

```diff
@@ -99,19 +139,42 @@ function test_fdm_hessians(fdm_backend)
     @test valscalar == fgrad(xvec, yvec)
     @test norm.(grad .- AD.gradient(fdm_backend, fhess, xvec)) == (0,)
     @test norm.(hess4 .- hess1) == (0,)
+    @test dfgraddxdx(xvec,yvec) ≈ hess1[1] atol=1e-10
+    @test xvec == xvec2
+    @test yvec == yvec2
+    fhess2 = x-> dfgraddx(x, yvec)
+    hess5 = AD.jacobian(fdm_backend, fhess2, xvec)
+    @test minimum(isapprox.(hess5, hess1, atol=1e-10))
 end
 
 function test_fdm_jvp(fdm_backend)
     v = (rand(length(xvec)), rand(length(yvec)))
-    pf1 = AD.pushforward_function(fdm_backend, fjac, xvec, yvec)(v)
+
+    if fdm_backend isa FDMBackend2 # augmented version of v
```

Comment:
Could you explain this part?

Reply:
This is related to the comment above: https://github.com/JuliaDiff/AbstractDifferentiation.jl/pull/2#issuecomment-839138438 . 

The resulting error traces back to the following: When we use: `v = (rand(length(xvec)), rand(length(yvec)))` with our `FDMBackend2` as the vector that we'd like to pushforward through a  function `fvec(x,y)` that takes two inputs `x` and `y`, we get out an output that corresponds to the directional derivative with "`v[1]` as the direction" along x *and* "`v[2]` as the direction" along y. The other backends however interprete the `v` as change `x` along the direction `v[1]` while keeping `y` fix and  change `y` in the direction `v[2]` while keeping `x` fix.